sdks/python/apache_beam/examples/complete/game/leader_board.py

```diff
@@ -0,0 +1,224 @@
+                                           trigger.AfterCount(20)),
+            accumulation_mode=trigger.AccumulationMode.ACCUMULATING)
+        | 'ExtractTeamScores' >> beam.Map(
+            lambda elem: (elem['team'], elem['score']))
+        | 'SumTeamScores' >> beam.CombinePerKey(sum)
+    )
+
+
+class CalculateUserScoreSums(beam.PTransform):
+  """Extract user/score pairs from the event stream using processing time, via
+  global windowing.
+
+  Get periodic updates on all users' running scores.
+  """
+  def __init__(self, allowed_lateness):
+    super(CalculateUserScoreSums, self).__init__()
+    self.allowed_lateness = allowed_lateness * 60
+
+  def expand(self, pcoll):
+    return (
+        # TODO: allowed_lateness not implemented yet in FixedWindows
+        # TODO: AfterProcessingTime not implemented yet, replace AfterCount
+        pcoll
+        | 'LeaderboardUserGlobalWindows' >> beam.WindowInto(
+            beam.window.GlobalWindows(),
+            trigger=trigger.Repeatedly(trigger.AfterCount(10)),
+            accumulation_mode=trigger.AccumulationMode.ACCUMULATING)
+        | 'ExtractUserScores' >> beam.Map(
+            lambda elem: (elem['user'], elem['score']))
+        | 'SumUserScores' >> beam.CombinePerKey(sum)
+    )
+
+
+def main():
+  """Main entry point; defines and runs the hourly_team_score pipeline."""
+  parser = argparse.ArgumentParser()
+
+  parser.add_argument('--topic',
+                      type=str,
+                      required=True,
+                      help='Pub/Sub topic to read from')
+  parser.add_argument('--output',
+                      type=str,
+                      required=True,
+                      help='Path to the output file(s).')
+  parser.add_argument('--team_window_duration',
+                      type=int,
+                      default=60,
+                      help='Numeric value of fixed window duration for team '
+                           'analysis, in minutes')
+  parser.add_argument('--allowed_lateness',
+                      type=int,
+                      default=120,
+                      help='Numeric value of allowed data lateness, in minutes')
+
+  args, pipeline_args = parser.parse_known_args()
+
+  # Enforce that this pipeline is always run in streaming mode
+  pipeline_args += ['--streaming']
+
+  with beam.Pipeline(argv=pipeline_args) as p:
+    events = (
+        p
+        | 'ReadPubSub' >> beam.io.gcp.pubsub.ReadStringsFromPubSub(args.topic)
+        | 'ParseGameEvent' >> beam.FlatMap(parse_game_event)
+        | 'AddEventTimestamps' >> beam.Map(
+            lambda elem: beam.window.TimestampedValue(elem, elem['timestamp']))
+    )
+
+    (events  # pylint: disable=expression-not-assigned
+     | 'CalculateTeamScoreSums' >> CalculateTeamScoreSums(
+         args.team_window_duration, args.allowed_lateness)
+     | 'FormatTeamScores' >> beam.ParDo(FormatTeamScores())
+     | 'WriteTeamScoreSums' >> beam.io.WriteToText(args.output+'-team_scores')
```

Comment:
Consider writing to BQ. 

DataflowRunner cannot write to text in streaming mode. What happens with DirectRunner?